pylint_custom/assert_introspection.py

```diff
@@ -0,0 +1,41 @@
+import astroid
+
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+class AssertIntrospectionChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "assert-introspection"
+    priority = -1
+    msgs = {
+        "W9901": (
+            "Assert statement has a failure message. This breaks pytest assert introspection.",
+            "assert-message-present",
+            "Asserts in tests should not have a fail message.",
+        ),
+    }
+
+    def __init__(self, linter=None):
+        super(AssertIntrospectionChecker, self).__init__(linter)
+        self._module_stack = []
+
+    def visit_module(self, node):
+        self._module_stack.append(is_a_test(node))
```

Comment:
@kylelaker-ccpo Just your own words coming back to you...
```suggestion
    def visit_module(self, node):
        """
        Tracks whether the module we're "in" is a "test" module.
        """
        self._module_stack.append(is_a_test(node))
```

Reply:
Rewrote the function and made the comment slightly different. I believe I met the spirit of this suggestion.